ui/src/main/java/com/hyperwallet/android/ui/view/widget/DateUtil.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * The MIT License (MIT)
+ * Copyright (c) 2018 Hyperwallet Systems Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package com.hyperwallet.android.ui.view.widget;
+
+import android.text.TextUtils;
+import android.text.format.DateFormat;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Locale;
+
+/**
+ * Class is used for manage and convert date {@link DateWidget}
+ */
+public final class DateUtil {
+
+    private static final String SERVER_DATE_PATTERN = "yyyy-MM-dd";
+    private static final String WIDGET_DATE_PATTERN = "dd MMMM yyyy";
+
+    private final SimpleDateFormat mServerDateFormat = new SimpleDateFormat(SERVER_DATE_PATTERN, Locale.getDefault());
+    private final SimpleDateFormat mWidgetDateFormat;
+
+    public DateUtil() {
+        mWidgetDateFormat = new SimpleDateFormat(
+                DateFormat.getBestDateTimePattern(Locale.getDefault(), WIDGET_DATE_PATTERN), Locale.getDefault());
+    }
+
+    /**
+     * Convert Date from server format (yyyy-MM-dd) to DateWidget format build by BestDateTimePattern @see {@link
+     * DateFormat}  or (dd MMMM yyyy)
+     *
+     * @param serverDate Date from server
+     * @return String date in the format by BestDateTimePattern or dd MMMM yyyy otherwise
+     * @throws DateParseException unable to convert server date to widget format
+     */
+    @NonNull
+    String convertDateFromServerToWidgetFormat(@Nullable final String serverDate) throws DateParseException {
+        try {
+            if (isServerDateValid(serverDate)) {
+                final Calendar calendar = Calendar.getInstance();
+                calendar.setTime(mServerDateFormat.parse(serverDate));
+                return mWidgetDateFormat.format(calendar.getTime());
+            }
+        } catch (NumberFormatException | ParseException | IndexOutOfBoundsException e) {
```

Comment:
what would be the use case where we could have a IndexOutOfBoundsException or NumberFormatException?
unless there's not a valid case, we can just throw the ParseException as part of the method signature